### PR TITLE
feat: added semantic versioning

### PR DESCRIPTION
**Pull request by csaw-nhs**
## Description

Added semantic versioning and release to this repo

## Context

This will version the repo and create release notes based on the commit history using SemVer conventions

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/workflows/cicd-2-publish.yaml [hunk 1]
```diff
@@ -63,41 +62,36 @@ jobs:
...
     needs: [publish]
     steps:
       - name: "Check prerequisites for notification"
         id: check
-        run: echo "secret_exist=${{ secrets.TEAMS_WEBHOOK_URL != '' }}" >> $GITHUB_OUTPUT
+        run: echo "secret_exist=${{ secrets.TEAMS_NOTIFICATION_WEBHOOK_URL != '' }}" >> $GITHUB_OUTPUT
       - name: "Notify on publishing packages"
         if: steps.check.outputs.secret_exist == 'true'
-        uses: nhs-england-tools/notify-msteams-action@v${{ needs.metadata.outputs.version }}
+        uses: nhs-england-tools/notify-msteams-action@v0.0.4
```
Comment (walteck) [inline .github/workflows/cicd-2-publish.yaml:91]: Could we set this to @latest here please?

File: .github/workflows/cicd-2-publish.yaml [hunk 2]
```diff
@@ -1,16 +1,17 @@
 name: "CI/CD publish"
 
 on:
-  pull_request:
-    types: [closed]
-    branches:
-      - main
+  push:
+    branches: [ "main" ]
+  # pull_request:
```
Comment (walteck) [inline .github/workflows/cicd-2-publish.yaml:6]: where these are commented out could we just delete the lines - they will still exist in git history if we need to refer back to them

**Review by walteck — CHANGES_REQUESTED** [review]
Looks good thanks, just a couple of comments

File: .github/workflows/cicd-2-publish.yaml (hunk 1 above)
Comment (csaw-nhs) [reply .github/workflows/cicd-2-publish.yaml:91]: I'll change it to user the main branch (latest isn't supported) https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-versioned-actions

**Review by walteck — APPROVED** [review]

**Pull request merged**